File: tests/machine/x/fs/save_jsonl_stdout.fs
open System
open System.Text.Json

exception Break
exception Continue

type Anon1 = {
    name: string
    age: int
}
let people = [{ name = "Alice"; age = 30 }; { name = "Bob"; age = 25 }]
(List.iter (fun row -> printfn "%s" (JsonSerializer.Serialize(row))) people)

File: tests/transpiler/x/clj/save_jsonl_stdout.clj
(ns main)

(require 'clojure.set)

(defn json-str [x]
  (cond (map? x) (str "{" (clojure.string/join "," (map (fn [[k v]] (str "\"" (name k) "\":" (json-str v))) x)) "}") (sequential? x) (str "[" (clojure.string/join "," (map json-str x)) "]") (string? x) (pr-str x) :else (str x)))

(defrecord Anon1 [name age])

(def people [{:name "Alice" :age 30} {:name "Bob" :age 25}])

(defn -main []
  (doseq [item people] (println (json-str item))))

(-main)
